### ISSUE TITLE
Best practice to convert AnyProperty -> SignalProducer -> AnyProperty

### ISSUE THREAD

**Issue by danishin** [ReactiveCocoa/ReactiveCocoa#2607]
I think I read an issue related to this a while back but I cannot find it anymore...

I'm using MVVM pattern and I have a protocol `TableViewModellable`:
```
protocol TableViewModellable {
  var count: AnyProperty<Int> { get } 
}
```
Then I'm deriving some convenience properties from the primitives like:
```
extension TableViewModellable {
  var isEmpty: SignalProducer<Bool, NoError> { return count.producer.map{ $0 == 0 }.skipRepeats() }
}
```
I'd love to make `isEmpty` `AnyProperty` like all the other properties that I'm exposing. I know I could do something like this:
```
extension SignalProducerType where Error == NoError {
  func toProperty() { 
    let property = MutableProperty<Value>
    property <~ self
    return property.immutable // immutable just returns `AnyProperty`
  }
}
```
But I can't help but to think this is kind of awkward. Should I just expose as `SignalProducer` or is this an **okay** practice to do (in terms of performance / corner cases)? Or perhaps I should just get over with the details and get stuff done?

**Comment by NachoSoto** [comment]
What you describe is a real problem: **there is currently no way to compose `PropertyType`s**. I opened #2535 to talk about this.

Ideally Swift would allow us to make `PropertyType` conform to `SignalType` and `SignalProducerType` so that you can apply the existing operators, but it's not possible at the moment.

**Comment by danishin** [comment]
Yeap that was the issue I read previously. Thanks for the link. 

Perhaps we can make free function `lift` to bridge from `SignalType` operators like we do with `SignalProducerType`? 

With that aside, can you think of any downsides to do conversion like the above (`SignalProducerType` -> `PropertyType`) as a workaround for now?

**Comment by ikesyo** [comment]
Now we have the `PropertyType.map` operator implemented in #2788. That should be a help for this issue. Closing.
